docs/dev/developer-guides/bridging/l1-l2-interop.md

```diff
@@ -1,72 +1,76 @@
-# L1 / L2 Interoperability
+# L1 / L2 interoperability
 
-While most of the execution will happen on L2, some use cases require interoperability with the L1 chain. The main use cases are building complex bridges, maintaining governance smart contracts on one chain that govern contracts on other chains, etc.
+## Common use cases
 
-In addition, the L2 censorship resistance is derived from the underlying chain, so the ability to send messages from Ethereum to zkSync is an important part of the censorship-resistance mechanism called the [priority queue](#priority-queue).
+Many use cases require multi-layer interoperability, such as:
 
-Sending transactions from Ethereum to zkSync is done via the zkSync smart contract. It allows the sender to request transactions directly from L1. Thereby allowing the permissionless passing of any data from Ethereum into zkSync.
-[Read more](../bridging/l1-l2.md) about messaging from L1 to L2.
+- The network's censorship resistance.
+- Custom bridges.
+- Multi-layer governing smart contracts.
+- Multi-layer transfers.
 
-## Priority queue
+## L1 to L2 communication
 
-The goal of the priority queue is to provide a censorship-resistant way to interact with zkSync in case the operator becomes malicious or unavailable.
-The way the priority queue works in zkSync Era is very close to how it worked in the previous version of zkSync.
-For the full picture, we first present how the priority queue works on zkSync Lite.
-This gives the rationale for the new design of the priority queue for zkSync Era.
+L1 to L2 communication is governed by the [`IZkSync.sol`](https://github.com/matter-labs/v2-testnet-contracts/blob/b8449bf9c819098cc8bfee0549ff5094456be51d/l1/contracts/zksync/interfaces/IZkSync.sol#L4) inherited interfaces.
 
-### How it works in zkSync Lite
+:::tip
+- If you prefer to learn-by-doing, the [cross chain governance tutorial](../../tutorials/cross-chain-tutorial.md) is a practical example of layer interoperability.
+:::
 
-In the previous version of zkSync, we only had two operations that could be sent to zkSync from L1:
+### Gas estimation
 
-- `Deposit` to bridge funds from Ethereum to zkSync.
-- `FullExit` to bridges the funds back from Ethereum (this is essentially the same as `Withdraw` in zkSync Era).
+The system processes gas estimation for transactions implicitly. However, it is also possible to implement the gas estimation processes explicitly.
+
+:::tip L1 to L2 gas estimation for transactions
+- Gas is measured in amount and not ERC20 token value so actual costs can vary.
+- The transaction process requires the current L1 gas price, transaction base cost, and transaction gas limit which defines the minimum amount of gas a transaction requires.
```

Comment:
This, and the previous sentence, seem to simply state how things work on Ethereum. Is this the intent?

> gas limit which defines the minimum amount of gas a transaction requires.

gas limit limits the amount of gas the transaction will consume. So it's about the maximum gas, not the minimum

Reply:
I also thought maximum @barakshani but it is minimum @StanislavBreadless please confirm.

Reply:
@kailash-manasarovar please correct (here and in other places): gas limit isn't the minimum

Reply:
Wait, I thought we established that it was indeed the lower bound and than consequently additional values are added to it to ensure fee coverage?

Reply:
This is really tricky, when one dev disagrees with another on something so fundamental like this.

Reply:
@StanislavBreadless do you disagree with anything in my previous comment?

And the context is whether we should state that the gas estimation or gas limit is a limit from below

Reply:
@barakshani 

It attempts to return the minimal value which is required for the transaction to execute successfully. Sometimes we do not know for sure and return an estimate or roughly minimal value. If it did not at least try to return the minimal value for the transaction to succeed, we could always supply maximal possible gas limit : ) 

I understand that both you and me know what `gasLimit` is, I don't mind anything if the wording feels better for you

Reply:
your thoughts @barakshani .. just to say, I originally had it saying maximum, and it was technically edited by SME input to minimum, and I'm still struggling with minimal == maximum ...

Reply:
Again, there are 2 different values we discuss here:
- The answer of estimageGas
- gasLimit within the tx

The dispute is about the first, where I claim that no "minimum" (nor maximum) should be used in this context (also, if it was the minimum we wouldn't have "out of gas" errors, that shows that the tx needed more gas).

We can refrain from speaking about minimum/maximum, but we cannot say that gasLimit is the minimum that the tx "requires" -- it is not, and it is the maximum the tx can consume. There's no disagreement between the two of us about it.

Reply:
lgtm. thanks!